wagtail/images/templatetags/wagtailimages_tags.py

```diff
@@ -1,12 +1,13 @@
 import re
+from functools import cache
```

Comment:
Looks like `functools.cache` was [only added in Python 3.9](https://docs.python.org/3/library/functools.html#functools.cache), so we can't use it in Wagtail until 3.8 is EOL.